cloud/aws/benchmarks/benchmarks.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+# The caller should download the static-tests and extract them into the
+# local directory.
+
+set -x
+
+TEST_FLAGS="-test.bench=.* -test.benchmem -test.run=NONE -test.count=10"
```

Comment:
true. the number of iterations is pretty close.

Here's the `benchstat <benchmark with --benchmem> <benchmark without --benchmem>` for all tests.

It's pretty close, usually in single digit percentage difference.


```
name                                  old time/op    new time/op     delta
RPCSSLClientScan1Version1Row-8           139µs ± 3%      149µs ±17%     ~     (p=0.912 n=10+10)
RPCSSLClientScan1Version10Rows-8         437µs ± 3%      424µs ± 2%   -3.15%   (p=0.001 n=10+8)
RPCSSLClientScan1Version100Rows-8       3.22ms ± 4%     3.19ms ± 4%     ~     (p=0.529 n=10+10)
RPCSSLClientScan1Version1000Rows-8      29.1ms ± 4%     29.5ms ± 5%     ~     (p=0.353 n=10+10)
RPCNoSSLClientScan1Version1Row-8         114µs ± 3%      120µs ±39%     ~     (p=0.143 n=10+10)
RPCNoSSLClientScan1Version10Rows-8       222µs ± 9%      200µs ± 2%   -9.76%   (p=0.000 n=10+9)
RPCNoSSLClientScan1Version100Rows-8     1.08ms ±16%     1.00ms ±17%     ~      (p=0.053 n=9+10)
RPCNoSSLClientScan1Version1000Rows-8    7.38ms ±15%     6.45ms ±14%  -12.71%  (p=0.003 n=10+10)

name                                  old speed      new speed       delta
RPCSSLClientScan1Version1Row-8        7.39MB/s ± 3%   6.94MB/s ±16%     ~     (p=0.912 n=10+10)
RPCSSLClientScan1Version10Rows-8      23.4MB/s ± 3%   24.2MB/s ± 2%   +3.24%   (p=0.001 n=10+8)
RPCSSLClientScan1Version100Rows-8     31.9MB/s ± 4%   32.1MB/s ± 4%     ~     (p=0.516 n=10+10)
RPCSSLClientScan1Version1000Rows-8    35.2MB/s ± 4%   34.7MB/s ± 5%     ~     (p=0.353 n=10+10)
RPCNoSSLClientScan1Version1Row-8      8.99MB/s ± 3%   8.74MB/s ±30%     ~     (p=0.143 n=10+10)
RPCNoSSLClientScan1Version10Rows-8    46.2MB/s ± 9%   51.1MB/s ± 2%  +10.62%   (p=0.000 n=10+9)
RPCNoSSLClientScan1Version100Rows-8   92.3MB/s ±26%  103.6MB/s ±15%  +12.30%  (p=0.029 n=10+10)
RPCNoSSLClientScan1Version1000Rows-8   140MB/s ±14%    160MB/s ±14%  +14.57%  (p=0.003 n=10+10)
name                   old time/op  new time/op  delta
Bank-8                  394µs ± 3%   365µs ± 6%   -7.56%  (p=0.000 n=10+10)
Select1_HTTP-8          474µs ±19%   369µs ±18%  -22.16%   (p=0.000 n=9+10)
Select1_HTTPS-8         474µs ± 3%   402µs ± 7%  -15.32%   (p=0.000 n=9+10)
Select1_RPC-8          95.8µs ± 3%  96.4µs ± 2%     ~      (p=0.497 n=10+9)
Select1_RPCS-8          135µs ± 8%   126µs ± 7%   -6.80%  (p=0.001 n=10+10)
Select1_Cockroach-8    89.5µs ± 3%  89.3µs ± 6%     ~     (p=0.436 n=10+10)
Insert1_Cockroach-8    1.31ms ± 1%  1.28ms ± 1%   -2.48%  (p=0.000 n=10+10)
Insert10_Cockroach-8   1.78ms ± 1%  1.73ms ± 2%   -3.12%  (p=0.000 n=10+10)
Insert100_Cockroach-8  5.95ms ± 1%  5.68ms ± 0%   -4.44%   (p=0.000 n=10+8)
Scan1_Cockroach-8       486µs ± 1%   476µs ± 2%   -2.01%    (p=0.001 n=9+9)
Scan10_Cockroach-8      572µs ± 1%   596µs ±10%     ~     (p=1.000 n=10+10)
Scan100_Cockroach-8    1.08ms ± 1%  1.05ms ± 1%   -2.94%   (p=0.000 n=10+9)
name               old time/op    new time/op    delta
EchoGobRPC1K-8       11.3µs ± 2%    11.3µs ± 1%   ~     (p=0.724 n=10+10)
EchoGobRPC64K-8       128µs ± 1%     129µs ± 1%   ~      (p=0.360 n=10+8)
EchoProtoRPC1K-8     9.67µs ± 1%    9.67µs ± 1%   ~      (p=0.762 n=10+8)
EchoProtoRPC64K-8     103µs ± 2%     102µs ± 2%   ~     (p=0.529 n=10+10)

name               old speed      new speed      delta
EchoGobRPC1K-8      181MB/s ± 2%   181MB/s ± 1%   ~     (p=0.739 n=10+10)
EchoGobRPC64K-8    1.02GB/s ± 1%  1.02GB/s ± 1%   ~      (p=0.360 n=10+8)
EchoProtoRPC1K-8    212MB/s ± 0%   212MB/s ± 1%   ~       (p=0.541 n=9+8)
EchoProtoRPC64K-8  1.28GB/s ± 2%  1.28GB/s ± 2%   ~     (p=0.529 n=10+10)
name                                old time/op  new time/op  delta
SingleRoundtripTxnWithLatency_0-8    962µs ± 2%   945µs ± 1%  -1.84%  (p=0.000 n=10+10)
SingleRoundtripTxnWithLatency_10-8  10.7ms ± 1%  10.5ms ± 1%  -1.12%  (p=0.000 n=10+10)
name                           old time/op  new time/op  delta
SubmitCommand1Node1Pending-8   20.7µs ± 2%  20.5µs ± 2%    ~     (p=0.159 n=10+10)
SubmitCommand3Nodes1Pending-8   124µs ± 1%   122µs ± 0%  -1.76%   (p=0.000 n=10+9)
SubmitCommand5Nodes1Pending-8   205µs ± 1%   201µs ± 1%  -2.03%   (p=0.000 n=10+9)
SubmitCommand1Node2Pending-8   15.1µs ± 1%  15.0µs ± 1%  -0.82%  (p=0.015 n=10+10)
SubmitCommand3Nodes2Pending-8  95.1µs ± 0%  93.5µs ± 1%  -1.61%    (p=0.000 n=6+9)
SubmitCommand5Nodes2Pending-8   153µs ± 1%   152µs ± 1%  -1.19%  (p=0.000 n=10+10)
SubmitCommand1Node4Pending-8   12.7µs ± 3%  12.7µs ± 1%    ~     (p=0.172 n=10+10)
SubmitCommand3Nodes4Pending-8  73.9µs ± 1%  72.8µs ± 1%  -1.52%  (p=0.000 n=10+10)
SubmitCommand5Nodes4Pending-8   118µs ± 1%   116µs ± 1%  -1.74%   (p=0.000 n=10+9)
name                                   old time/op    new time/op     delta
MVCCStats-8                              6.60µs ± 1%     6.58µs ± 1%     ~     (p=0.324 n=10+10)
MVCCScan1Version1Row8Bytes-8             5.34µs ± 0%     5.15µs ± 2%   -3.46%    (p=0.000 n=9+9)
MVCCScan1Version1Row64Bytes-8            7.11µs ± 1%     6.62µs ± 3%   -6.83%    (p=0.000 n=9+9)
MVCCScan1Version1Row512Bytes-8           13.8µs ± 1%     13.2µs ± 3%   -4.02%    (p=0.000 n=9+9)
MVCCScan1Version10Rows8Bytes-8           28.4µs ± 0%     27.5µs ± 0%   -2.92%    (p=0.000 n=9+8)
MVCCScan1Version10Rows64Bytes-8          32.6µs ± 1%     31.2µs ± 1%   -4.15%    (p=0.000 n=9+9)
MVCCScan1Version10Rows512Bytes-8         62.1µs ± 1%     60.2µs ± 2%   -2.99%  (p=0.000 n=10+10)
MVCCScan1Version100Rows8Bytes-8           241µs ± 0%      235µs ± 0%   -2.29%  (p=0.000 n=10+10)
MVCCScan1Version100Rows64Bytes-8          267µs ± 0%      261µs ± 1%   -2.25%  (p=0.000 n=10+10)
MVCCScan1Version100Rows512Bytes-8         515µs ± 1%      500µs ± 1%   -2.89%  (p=0.000 n=10+10)
MVCCScan1Version1000Rows8Bytes-8         2.33ms ± 0%     2.28ms ± 1%   -2.35%   (p=0.000 n=10+9)
MVCCScan1Version1000Rows64Bytes-8        2.57ms ± 0%     2.50ms ± 1%   -2.76%   (p=0.000 n=8+10)
MVCCScan1Version1000Rows512Bytes-8       5.00ms ± 1%     4.84ms ± 1%   -3.24%  (p=0.000 n=10+10)
MVCCScan10Versions1Row8Bytes-8           11.2µs ± 1%     10.5µs ± 1%   -5.82%    (p=0.000 n=9+9)
MVCCScan10Versions1Row64Bytes-8          13.5µs ± 1%     13.0µs ± 4%   -3.81%    (p=0.000 n=9+9)
MVCCScan10Versions1Row512Bytes-8         61.1µs ± 1%     56.3µs ± 2%   -7.85%  (p=0.000 n=10+10)
MVCCScan10Versions10Rows8Bytes-8         44.6µs ± 0%     43.5µs ± 1%   -2.54%  (p=0.000 n=10+10)
MVCCScan10Versions10Rows64Bytes-8        54.9µs ± 0%     53.6µs ± 1%   -2.25%  (p=0.000 n=10+10)
MVCCScan10Versions10Rows512Bytes-8        216µs ± 4%      208µs ± 2%   -4.02%   (p=0.000 n=10+8)
MVCCScan10Versions100Rows8Bytes-8         353µs ± 1%      342µs ± 1%   -2.93%  (p=0.000 n=10+10)
MVCCScan10Versions100Rows64Bytes-8        438µs ± 1%      426µs ± 1%   -2.82%  (p=0.000 n=10+10)
MVCCScan10Versions100Rows512Bytes-8      1.66ms ± 3%     1.62ms ± 2%   -2.77%   (p=0.000 n=10+9)
MVCCScan10Versions1000Rows8Bytes-8       3.39ms ± 1%     3.30ms ± 1%   -2.64%  (p=0.000 n=10+10)
MVCCScan10Versions1000Rows64Bytes-8      4.23ms ± 2%     4.09ms ± 1%   -3.25%  (p=0.000 n=10+10)
MVCCScan10Versions1000Rows512Bytes-8     16.3ms ± 5%     15.7ms ± 4%   -3.94%  (p=0.003 n=10+10)
MVCCScan100Versions1Row512Bytes-8        72.7µs ± 1%     85.7µs ± 3%  +17.86%    (p=0.000 n=9+9)
MVCCScan100Versions10Rows512Bytes-8       382µs ± 0%      435µs ± 3%  +13.86%  (p=0.000 n=10+10)
MVCCScan100Versions100Rows512Bytes-8     3.36ms ± 1%     3.86ms ± 1%  +14.92%    (p=0.000 n=9+7)
MVCCScan100Versions1000Rows512Bytes-8    33.0ms ± 4%     37.8ms ± 5%  +14.43%  (p=0.000 n=10+10)
MVCCGet1Version8Bytes-8                  19.1µs ± 8%     19.2µs ±10%     ~     (p=0.280 n=10+10)
MVCCGet10Versions8Bytes-8                27.7µs ± 4%     27.7µs ± 5%     ~      (p=0.633 n=8+10)
MVCCGet100Versions8Bytes-8               39.6µs ± 3%     39.9µs ±13%     ~      (p=0.573 n=8+10)
MVCCPut10-8                              5.15µs ± 1%     4.96µs ± 1%   -3.53%  (p=0.000 n=10+10)
MVCCPut100-8                             5.44µs ± 1%     5.27µs ± 0%   -3.07%   (p=0.000 n=10+9)
MVCCPut1000-8                            8.45µs ± 1%     8.14µs ± 1%   -3.73%  (p=0.000 n=10+10)
MVCCPut10000-8                           34.9µs ± 1%     33.4µs ± 3%   -4.24%  (p=0.000 n=10+10)
MVCCBatch1Put10-8                        6.37µs ± 1%     6.16µs ± 0%   -3.29%  (p=0.000 n=10+10)
MVCCBatch100Put10-8                      4.39µs ± 0%     4.30µs ± 1%   -2.20%   (p=0.000 n=9+10)
MVCCBatch10000Put10-8                    4.81µs ± 1%     4.72µs ± 0%   -1.94%    (p=0.000 n=9+9)
MVCCBatch100000Put10-8                   4.95µs ± 1%     4.86µs ± 2%   -1.90%   (p=0.000 n=10+9)
MVCCMergeTimeSeries-8                    8.58µs ± 6%     8.30µs ± 6%     ~     (p=0.123 n=10+10)
MVCCDeleteRange1Version8Bytes-8           117ms ± 0%      126ms ± 1%   +8.29%   (p=0.000 n=9+10)
MVCCDeleteRange1Version32Bytes-8         80.2ms ± 1%     88.4ms ± 8%  +10.12%  (p=0.000 n=10+10)
MVCCDeleteRange1Version256Bytes-8        23.1ms ± 0%     22.6ms ± 1%   -2.24%   (p=0.000 n=8+10)
MVCCComputeStats1Version8Bytes-8          262ms ± 1%      257ms ± 2%   -2.05%    (p=0.000 n=9+9)
MVCCComputeStats1Version32Bytes-8         221ms ± 1%      213ms ± 3%   -3.68%    (p=0.000 n=9+9)
MVCCComputeStats1Version256Bytes-8        139ms ± 1%      131ms ± 1%   -5.97%    (p=0.000 n=9+9)

name                                   old speed      new speed       delta
MVCCStats-8                            15.8MB/s ± 1%   15.8MB/s ± 1%     ~     (p=0.305 n=10+10)
MVCCScan1Version1Row8Bytes-8           1.50MB/s ± 1%   1.56MB/s ± 1%   +3.91%    (p=0.000 n=9+8)
MVCCScan1Version1Row64Bytes-8          9.00MB/s ± 1%   9.66MB/s ± 3%   +7.35%    (p=0.000 n=9+9)
MVCCScan1Version1Row512Bytes-8         37.1MB/s ± 1%   38.7MB/s ± 3%   +4.22%    (p=0.000 n=9+9)
MVCCScan1Version10Rows8Bytes-8         2.82MB/s ± 0%   2.90MB/s ± 1%   +2.84%    (p=0.000 n=6+9)
MVCCScan1Version10Rows64Bytes-8        19.6MB/s ± 1%   20.5MB/s ± 1%   +4.33%    (p=0.000 n=9+9)
MVCCScan1Version10Rows512Bytes-8       82.5MB/s ± 1%   85.0MB/s ± 2%   +3.09%  (p=0.000 n=10+10)
MVCCScan1Version100Rows8Bytes-8        3.33MB/s ± 0%   3.40MB/s ± 0%   +2.35%  (p=0.000 n=10+10)
MVCCScan1Version100Rows64Bytes-8       24.0MB/s ± 0%   24.6MB/s ± 1%   +2.30%  (p=0.000 n=10+10)
MVCCScan1Version100Rows512Bytes-8      99.4MB/s ± 1%  102.3MB/s ± 1%   +2.98%  (p=0.000 n=10+10)
MVCCScan1Version1000Rows8Bytes-8       3.43MB/s ± 0%   3.52MB/s ± 1%   +2.40%   (p=0.000 n=10+9)
MVCCScan1Version1000Rows64Bytes-8      24.9MB/s ± 0%   25.6MB/s ± 1%   +2.84%   (p=0.000 n=8+10)
MVCCScan1Version1000Rows512Bytes-8      102MB/s ± 1%    106MB/s ± 1%   +3.34%  (p=0.000 n=10+10)
MVCCScan10Versions1Row8Bytes-8          716kB/s ± 1%    760kB/s ± 0%   +6.21%    (p=0.000 n=9+7)
MVCCScan10Versions1Row64Bytes-8        4.73MB/s ± 1%   4.92MB/s ± 4%   +4.04%    (p=0.000 n=9+9)
MVCCScan10Versions1Row512Bytes-8       8.38MB/s ± 1%   9.10MB/s ± 2%   +8.54%  (p=0.000 n=10+10)
MVCCScan10Versions10Rows8Bytes-8       1.79MB/s ± 0%   1.84MB/s ± 1%   +2.56%  (p=0.000 n=10+10)
MVCCScan10Versions10Rows64Bytes-8      11.7MB/s ± 0%   11.9MB/s ± 1%   +2.31%  (p=0.000 n=10+10)
MVCCScan10Versions10Rows512Bytes-8     23.7MB/s ± 4%   24.7MB/s ± 2%   +4.16%   (p=0.000 n=10+8)
MVCCScan10Versions100Rows8Bytes-8      2.27MB/s ± 1%   2.34MB/s ± 0%   +3.09%  (p=0.000 n=10+10)
MVCCScan10Versions100Rows64Bytes-8     14.6MB/s ± 1%   15.0MB/s ± 1%   +2.90%  (p=0.000 n=10+10)
MVCCScan10Versions100Rows512Bytes-8    30.8MB/s ± 3%   31.6MB/s ± 2%   +2.83%   (p=0.000 n=10+9)
MVCCScan10Versions1000Rows8Bytes-8     2.36MB/s ± 1%   2.42MB/s ± 1%   +2.67%  (p=0.000 n=10+10)
MVCCScan10Versions1000Rows64Bytes-8    15.1MB/s ± 2%   15.6MB/s ± 1%   +3.36%  (p=0.000 n=10+10)
MVCCScan10Versions1000Rows512Bytes-8   31.3MB/s ± 5%   32.6MB/s ± 4%   +4.07%  (p=0.003 n=10+10)
MVCCScan100Versions1Row512Bytes-8      7.04MB/s ± 1%   5.97MB/s ± 3%  -15.15%    (p=0.000 n=9+9)
MVCCScan100Versions10Rows512Bytes-8    13.4MB/s ± 0%   11.8MB/s ± 3%  -12.15%  (p=0.000 n=10+10)
MVCCScan100Versions100Rows512Bytes-8   15.2MB/s ± 1%   13.3MB/s ± 1%  -12.97%    (p=0.000 n=9+7)
MVCCScan100Versions1000Rows512Bytes-8  15.5MB/s ± 4%   13.6MB/s ± 5%  -12.57%  (p=0.000 n=10+10)
MVCCGet1Version8Bytes-8                 421kB/s ± 7%    418kB/s ± 9%     ~     (p=0.333 n=10+10)
MVCCGet10Versions8Bytes-8               289kB/s ± 3%    290kB/s ± 3%     ~      (p=0.969 n=8+10)
MVCCGet100Versions8Bytes-8              200kB/s ± 0%    206kB/s ± 3%   +3.12%    (p=0.037 n=7+8)
MVCCPut10-8                            1.94MB/s ± 1%   2.01MB/s ± 1%   +3.53%   (p=0.000 n=9+10)
MVCCPut100-8                           18.4MB/s ± 1%   19.0MB/s ± 0%   +3.15%   (p=0.000 n=10+9)
MVCCPut1000-8                           118MB/s ± 1%    123MB/s ± 1%   +3.88%  (p=0.000 n=10+10)
MVCCPut10000-8                          287MB/s ± 1%    300MB/s ± 3%   +4.46%  (p=0.000 n=10+10)
MVCCBatch1Put10-8                      1.57MB/s ± 0%   1.62MB/s ± 0%   +3.38%   (p=0.000 n=6+10)
MVCCBatch100Put10-8                    2.28MB/s ± 0%   2.32MB/s ± 1%   +2.17%   (p=0.000 n=9+10)
MVCCBatch10000Put10-8                  2.08MB/s ± 1%   2.12MB/s ± 0%   +2.03%    (p=0.000 n=9+8)
MVCCBatch100000Put10-8                 2.02MB/s ± 1%   2.06MB/s ± 2%   +1.87%   (p=0.000 n=10+9)
MVCCDeleteRange1Version8Bytes-8        4.50MB/s ± 1%   4.16MB/s ± 1%   -7.69%  (p=0.000 n=10+10)
MVCCDeleteRange1Version32Bytes-8       6.54MB/s ± 1%   5.95MB/s ± 7%   -9.03%  (p=0.000 n=10+10)
MVCCDeleteRange1Version256Bytes-8      22.7MB/s ± 0%   23.2MB/s ± 1%   +2.30%   (p=0.000 n=8+10)
MVCCComputeStats1Version8Bytes-8        256MB/s ± 1%    262MB/s ± 2%   +2.10%    (p=0.000 n=9+9)
MVCCComputeStats1Version32Bytes-8       303MB/s ± 1%    315MB/s ± 3%   +3.83%    (p=0.000 n=9+9)
MVCCComputeStats1Version256Bytes-8      483MB/s ± 1%    513MB/s ± 1%   +6.35%    (p=0.000 n=9+9)
name                             old time/op  new time/op  delta
WriteCmdNoEvents-8                146µs ± 2%   150µs ± 1%  +2.39%  (p=0.001 n=10+10)
WriteCmdWithEvents-8              144µs ± 1%   144µs ± 2%    ~      (p=0.720 n=10+9)
WriteCmdWithEventsAndConsumer-8   144µs ± 2%   145µs ± 3%    ~     (p=0.143 n=10+10)
name                      old time/op  new time/op  delta
EncodeUint32-8            14.4ns ± 0%  14.5ns ± 0%  +0.69%        (p=0.000 n=10+10)
DecodeUint32-8            17.7ns ± 0%  17.8ns ± 0%  +0.56%         (p=0.000 n=8+10)
EncodeUint64-8            16.7ns ± 0%  16.7ns ± 0%    ~           (p=0.211 n=10+10)
DecodeUint64-8            22.4ns ± 0%  22.4ns ± 0%    ~     (all samples are equal)
EncodeVarint-8            24.4ns ± 0%  24.4ns ± 0%    ~     (all samples are equal)
DecodeVarint-8            30.2ns ± 0%  30.2ns ± 0%    ~     (all samples are equal)
EncodeUvarint-8           22.1ns ± 0%  22.1ns ± 0%    ~     (all samples are equal)
DecodeUvarint-8           26.3ns ± 0%  26.3ns ± 0%    ~     (all samples are equal)
EncodeBytes-8             54.0ns ± 0%  54.1ns ± 0%  +0.19%        (p=0.000 n=10+10)
EncodeBytesDecreasing-8   77.9ns ± 0%  78.1ns ± 0%  +0.21%        (p=0.000 n=10+10)
DecodeBytes-8             88.0ns ± 0%  88.1ns ± 0%  +0.11%          (p=0.000 n=9+8)
DecodeBytesDecreasing-8    107ns ± 0%   108ns ± 0%  +0.93%         (p=0.000 n=10+9)
EncodeString-8            57.5ns ± 0%  57.6ns ± 0%  +0.17%          (p=0.000 n=9+9)
EncodeStringDecreasing-8  81.6ns ± 0%  81.8ns ± 0%  +0.18%        (p=0.000 n=10+10)
DecodeString-8             178ns ± 0%   178ns ± 1%    ~             (p=0.540 n=7+9)
DecodeStringDecreasing-8   194ns ± 1%   194ns ± 1%    ~           (p=0.378 n=10+10)
EncodeFloat-8              301ns ± 1%   300ns ± 1%    ~            (p=0.362 n=9+10)
DecodeFloat-8              234ns ± 0%   235ns ± 0%  +0.26%         (p=0.023 n=8+10)
name      old time/op  new time/op  delta
Header-8   341ns ± 0%   343ns ± 0%  +0.59%  (p=0.000 n=10+10)
```